### PR TITLE
ASimFileEventMicrosoftSysmon.yaml-27

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for WindowsEvent and added WindowsEvent parser in it. In Event file removed WindowsEvent parser and in ASimFileEvent added WindowsEvent Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the WindowsEven and Event parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check below failed KQL validation. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: File event ASIM parser for Windows Sysmon
-  Version: '0.4.1'
-  LastUpdated: Nov 14 2023
+  Version: '0.5.1'
+  LastUpdated: June 6 2024
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml:4]: date correction

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+Parser:
+  Title: File event ASIM parser for Windows Sysmon
+  Version: '0.4.1'
+  LastUpdated: June 6 2024
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml:4]: date correction

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: File event ASIM filtering parser for Windows Sysmon
-  Version: "0.4.1"
-  LastUpdated: Nov 14 2023
+  Version: "0.5.1"
+  LastUpdated: June 6 2024
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml:4]: date correction

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 4]
```diff
@@ -0,0 +1,168 @@
+Parser:
+  Title: File event ASIM filtering parser for Windows Sysmon
+  Version: "0.5.1"
+  LastUpdated: June 6 2024
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml:4]: date correction

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 8]
```diff
@@ -0,0 +1,168 @@
+Parser:
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml:1]: (1) Warning: Missing recommended field [EventUid]

File: Parsers/ASimFileEvent/test/Microsoft_Sysmon_ASimFileEvent_DataTest.csv [hunk 9]
```diff
@@ -0,0 +1,6 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 10 records (100.0%) for field [HashType] of type [Enumerated]: [""SHA256""] (Schema:FileEvent)"
```
Comment (vakohl) [inline Parsers/ASimFileEvent/test/Microsoft_Sysmon_ASimFileEvent_DataTest.csv:2]: can we check this error?

File: Parsers/ASimFileEvent/test/Microsoft_Sysmon_VimFileEvent_DataTest.csv [hunk 10]
```diff
@@ -0,0 +1,6 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 10 records (100.0%) for field [HashType] of type [Enumerated]: [""SHA256""] (Schema:FileEvent)"
```
Comment (vakohl) [inline Parsers/ASimFileEvent/test/Microsoft_Sysmon_VimFileEvent_DataTest.csv:2]: please check

File: Parsers/ASimFileEvent/test/Microsoft_Sysmon_Windows_Even_VimFileEvent_DataTest.csv [hunk 11]
```diff
@@ -0,0 +1,6 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 10 records (100.0%) for field [HashType] of type [Enumerated]: [""SHA256""] (Schema:FileEvent)"
```
Comment (vakohl) [inline Parsers/ASimFileEvent/test/Microsoft_Sysmon_Windows_Even_VimFileEvent_DataTest.csv:2]: please check

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml [hunk 12]
```diff
@@ -12,7 +12,8 @@ References:
   Link: https://aka.ms/ASimFileEventDoc
 - Title: ASIM
   Link: https://aka.ms/AboutASIM
-Description: This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, to the ASIM file event schema.
+Description: 
+  This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, logs ingested in 'Event' table to the ASIM file event schema.
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml:16]: description doesn't seem right

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+Normalization:
+  Schema: FileEvent
+  Version: '0.2.1'
+References:
+- Title: ASIM File Event Schema
+  Link: https://aka.ms/ASimFileEventDoc
+- Title: ASIM
+  Link: https://aka.ms/AboutASIM
+Description: 
+  This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, logs ingested in 'WindowsEvent' table to the ASIM file event schema.
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml:16]: description doesn't seem right

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml [hunk 14]
```diff
@@ -12,7 +12,8 @@ References:
     Link: https://aka.ms/ASimFileEventDoc
   - Title: ASIM
     Link: https://aka.ms/AboutASIM
-Description: This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, to the ASIM file event schema.
+Description: 
+  This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, logs ingested in 'Event' table to the ASIM file event schema.
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml:16]: description doesn't seem right

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+Normalization:
+  Schema: FileEvent
+  Version: "0.2.1"
+References:
+  - Title: ASIM File Event Schema
+    Link: https://aka.ms/ASimFileEventDoc
+  - Title: ASIM
+    Link: https://aka.ms/AboutASIM
+Description: 
+  This ASIM parser supports normalizing Sysmon event 11, 23, and 26, stored in either the Event or WindowsEvent tables, logs ingested in 'WindowsEvent' table to the ASIM file event schema.
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml:16]: description doesn't seem right

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml (hunk 5 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml:1]: Done

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 6 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml:1]: Done

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml (hunk 7 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml:1]: Done

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 8 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml:1]: Done

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml (hunk 12 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmon.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 13 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml (hunk 14 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmon.yaml]: Done

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 15 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please work on open comment. Thanks

File: Parsers/ASimFileEvent/test/Microsoft_Sysmon_VimFileEvent_DataTest.csv (hunk 10 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/test/Microsoft_Sysmon_VimFileEvent_DataTest.csv:2]: added supported values in ARM tester

File: Parsers/ASimFileEvent/test/Microsoft_Sysmon_Windows_Even_VimFileEvent_DataTest.csv (hunk 11 above)
Comment (Alekhya0824) [reply Parsers/ASimFileEvent/test/Microsoft_Sysmon_Windows_Even_VimFileEvent_DataTest.csv:2]: added supported values in ARM tester

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+              EventStartTime = todatetime(EventData.UtcTime),
+              RuleName = tostring(EventData.RuleName),
+              Hashes = tostring(EventData.Hashes)
+          | parse ActingProcessGuid with "{" ActingProcessGuid "}"
+          | project-away EventData
+  };
+      WindowsEventParser 
+      | project-rename
+          DvcHostname = Computer,
+          //EventUid = _ItemId,
```
Comment (vakohl) [inline Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml:105]: please remove this

File: Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+             EventStartTime = todatetime(EventData.UtcTime),
+             RuleName = tostring(EventData.RuleName),
+             Hashes = tostring(EventData.Hashes)
+          | parse ActingProcessGuid with "{" ActingProcessGuid "}"
+          | project-away EventData
+       };
+       WindowsEventParser
+       | project-rename
+           DvcHostname = Computer,
+           //EventUid = _ItemId,
```
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/ASimFileEventMicrosoftSysmonWindowsEvent.yaml]: I've removed it.

File: Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml (hunk 16 above)
Comment (vakohl) [reply Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftSysmonWindowsEvent.yaml]: I've removed it.

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**